addons/grenades/functions/fnc_throwGrenade.sqf

```diff
@@ -29,21 +29,11 @@ private _config = configFile >> "CfgAmmo" >> _ammo;
 
 // handle special grenades and sounds
 if (local _unit) then {
-    // handle priming sound, if not IR grenade
-    if (_ammo isKindOf "GrenadeHand") then {
-        private _posASLProjectile = getPosASL _projectile;
-        
-        switch (true) do {
-            case (_ammo isKindOf "Chemlight_base"): {
-                playSound3D ["A3\sounds_f\weapons\Other\dry4.wss", objNull, false, _posASLProjectile, 3, 2, 10]
-            };
-            case (_ammo isKindOf "ACE_G_Handflare_White"): {
-                playSound3D ["A3\sounds_f\weapons\smokeshell\smoke_1.wss", objNull, false, _posASLProjectile, 0.4, 1, 10]
-            };
-            default {
-                playSound3D ["A3\sounds_f\weapons\grenades\Grenade_PullPin.wss", objNull, false, _posASLProjectile, 1.5, 1, 10]
-            };
-        };
+    // handle priming sound, if present
+    private _soundConfig = getArray (configFile >> "CfgAmmo" >> _ammo >> QGVAR(pullPinSound));
+    if (count _soundConfig > 0) then {
+        _soundConfig params ["_file", "_volume", "_pitch", "_distance"];
```

Comment:
There are no default values in `class Default` (and I generally don't think it's a good way of doing things for addons, because of potential loading order issues).
Would this error if a 3rd party modded grenade is used that doesnt have this array defined?
`getArray` would report `[]` and this `params` would make all variabled undefined.